zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java

```diff
@@ -148,13 +151,24 @@ public BulkCallBuilder(ElasticsearchStorage es, float esVersion, String tag) {
       return headers;
     }
 
-    @Override public void writeBody(HttpCall.RequestStream requestStream) {
-      for (IndexEntry<?> entry : entries) {
-        if (!requestStream.tryWrite(HttpData.wrap(serialize(alloc, entry, shouldAddType)))) {
-          // Stream aborted, no need to serialize anymore.
-          return;
-        }
+    @Override
+    public HttpRequest get() {
+      final HttpRequestWriter writer = HttpRequest.streaming(headers);
+      writeEntry(writer, 0);
+      return writer;
+    }
+
+    // Use backpressure, so it's not buffering at all.
```

Comment:
Most people will barely understand this, so offering the long form comment directly :)

```suggestion
    // There's a high chance that the response is received before the request is complete.
    // This can be a problem for BulkCallBuilder when it's sending streaming requests.
    // Hence, we use backpressure, so it's not buffering at all.
```

Reply:
@adriancole After I changed it, it's not working because there's a still chance that the response is received before the request is fully sent. 😅 
Let me just roll back and fix it in Armeria. Sorry about it. 😅

Reply:
ok we can keep this open as it sounds like even if we are seeing this in tests, it could happen in real life. If the comment drifts later, then we update it along with the armeria release.

We were planning to do a release in a few days, but I think this is a good reason to wait a week if it takes that to sort it upstream.